File: frontend/src/services/workspaceService.ts
import apiClient from './api'
import type { ApiResponse, Workspace } from '@/types'

export const workspaceService = {
  getWorkspaces: async (): Promise<ApiResponse<Workspace[]>> => {
    const response = await apiClient.get<ApiResponse<Workspace[]>>('/workspaces')
    return response.data
  },

  getWorkspaceById: async (id: number): Promise<ApiResponse<Workspace>> => {
    const response = await apiClient.get<ApiResponse<Workspace>>(`/workspaces/${id}`)
    return response.data
  },

  createWorkspace: async (data: { name: string; description: string }): Promise<ApiResponse<Workspace>> => {
    const response = await apiClient.post<ApiResponse<Workspace>>('/workspaces', data)
    return response.data
  },

  updateWorkspace: async (id: number, data: { name: string; description: string }): Promise<ApiResponse<Workspace>> => {
    const response = await apiClient.put<ApiResponse<Workspace>>(`/workspaces/${id}`, data)
    return response.data
  },

  deleteWorkspace: async (id: number): Promise<ApiResponse<void>> => {
    const response = await apiClient.delete<ApiResponse<void>>(`/workspaces/${id}`)
    return response.data
  },
}
export default workspaceService

File: backend/src/main/java/com/semanticvault/service/WorkspaceService.java
package com.semanticvault.service;

import com.semanticvault.dto.WorkspaceDto;
import com.semanticvault.model.Workspace;
import com.semanticvault.repository.WorkspaceRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class WorkspaceService {

    private final WorkspaceRepository workspaceRepository;

    public WorkspaceService(WorkspaceRepository workspaceRepository) {
        this.workspaceRepository = workspaceRepository;
    }

    public List<Workspace> getAllWorkspaces() {
        return workspaceRepository.findAll();
    }

    public Workspace getWorkspaceById(Long id) {
        return workspaceRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Workspace not found with ID: " + id));
    }

    public Workspace createWorkspace(WorkspaceDto dto) {
        Workspace workspace = Workspace.builder()
                .name(dto.getName())
                .description(dto.getDescription())
                .createdAt(LocalDateTime.now())
                .updatedAt(LocalDateTime.now())
                .totalDocuments(0)
                .totalVectors(0)
                .build();
        return workspaceRepository.save(workspace);
    }

    public Workspace updateWorkspace(Long id, WorkspaceDto dto) {
        Workspace existing = getWorkspaceById(id);
        existing.setName(dto.getName());
        existing.setDescription(dto.getDescription());
        existing.setUpdatedAt(LocalDateTime.now());
        return workspaceRepository.save(existing);
    }

    public void deleteWorkspace(Long id) {
        if (!workspaceRepository.existsById(id)) {
            throw new IllegalArgumentException("Workspace not found with ID: " + id);
        }
        workspaceRepository.deleteById(id);
    }
}
